src/NDItk/multigroup/FissionNeutronSpectrumMatrix/test/FissionNeutronSpectrumMatrix.test.cpp

```diff
@@ -0,0 +1,235 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+using Catch::Matchers::WithinRel;
+
+// what we are testing
+#include "NDItk/multigroup/FissionNeutronSpectrumMatrix.hpp"
+
+// other includes
+
+// convenience typedefs
+using namespace njoy::NDItk;
+using FissionNeutronSpectrumMatrix = multigroup::FissionNeutronSpectrumMatrix;
+using FissionType = multigroup::FissionType;
+
+std::string chunk();
+void verifyChunk( const FissionNeutronSpectrumMatrix& );
+std::string chunkWithInsufficientNumberValues();
+
+SCENARIO( "FissionNeutronSpectrumMatrix" ) {
+
+  GIVEN( "valid data for a FissionNeutronSpectrumMatrix instance" ) {
+
+    std::string record = chunk();
+
+    WHEN( "the data is given explicitly" ) {
+
+      std::vector< double > values = {   1,   2,   3,   4,   5,   6,   7,
+                                       1.1, 2.1, 3.1, 4.1, 5.1, 6.1, 7.1,
+                                       1.2, 2.2, 3.2, 4.2, 5.2, 6.2, 7.2,
+                                       1.3, 2.3, 3.3, 4.3, 5.3, 6.3, 7.3,
+                                       1.4, 2.4, 3.4, 4.4, 5.4, 6.4, 7.4,
+                                       1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5,
+                                       1.6, 2.6, 3.6, 4.6, 5.6, 6.6, 7.6 };
+
+      FissionNeutronSpectrumMatrix chunk( FissionType::Prompt, std::move( values ) );
+
+      THEN( "a FissionNeutronSpectrumMatrix can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+
+    WHEN( "the data is read using iterators" ) {
+
+      auto iter = record.begin() + 10;
+      auto end = record.end();
+
+      FissionNeutronSpectrumMatrix chunk( FissionType::Prompt );
+      chunk.read( iter, end, 7 );
+
+      THEN( "a FissionNeutronSpectrumMatrix can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+
+  GIVEN( "invalid data for a FissionNeutronSpectrumMatrix instance" ) {
+
+    WHEN( "the number of weight values is insufficient" ) {
```

Comment:
Not weights.

Reply:
Also, we should test the "matrix not square" bit too.

Reply:
Damn you copy/paste!

Reply:
Changed in fix/review-part6. And added square test.